experimental/sdg-tracker/src/components/search/Search.tsx

```diff
@@ -245,22 +245,24 @@ const Search = () => {
     async (searchQuery?: string) => {
       setIsSearching(true);
       setErrorMessage("");
-      const responseObj = await dataCommonsClient.detect(searchQuery || query);
+      const fulfillResponse = await dataCommonsClient.detectAndFulfill(
+        searchQuery || query
+      );
       setIsSearching(false);
 
-      if (responseObj.failure) {
-        setErrorMessage(responseObj.failure);
+      if (fulfillResponse.failure || fulfillResponse.userMessage) {
```

Comment:
Jennifer should confirm, but I think this is how error is detected: https://github.com/datacommonsorg/website/blob/989710ebc78b643c0abee61333eba875702ee11a/static/js/apps/explore/app.tsx#L159-L167

userMessage can be valid even for a non-error case.

Reply:
yes, I also left a comment about this that failure and userMessage should be handled separately

Reply:
Separated the handling logic for userMessage and failure